src/rhsmlib/utils.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import print_function, division, absolute_import
+
+# Copyright (c) 2017 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+
+"""
+This module includes several utils that could be used by several client
+applications.
+"""
+
+
+class Singleton(object):
+    """
+    Singleton and parent for singletons
+    """
+    _instance = None
+    _initialized = False
```

Comment:
Where is this used?

Reply:
Good point. I realized that I do something wrong here and we will need to use it.

Reply:
So it is used in decorator `no_reinitialization`.